multinode_integration_tests/src/masq_mock_node.rs

```diff
@@ -145,43 +146,59 @@ impl MASQNode for MASQMockNode {
     }
 }
 
-pub struct ConfigurableMASQMockNode {
+pub struct MASQMockNodeWithMutableHandle {
     pub control_stream: RefCell<TcpStream>,
     configurable_guts: MASQMockNodeGuts,
 }
 
-impl ConfigurableMASQMockNode {
-    pub fn start_with_public_key(
-        ports: Vec<u16>,
-        index: usize,
-        host_node_parent_dir: Option<String>,
-        public_key: &PublicKey,
-        chain: Chain,
-    ) -> ConfigurableMASQMockNode {
-        let main_cryptde = CryptDENull::from(public_key, chain);
-        let mut key = public_key.as_slice().to_vec();
-        key.reverse();
-        let alias_cryptde = CryptDENull::from(&PublicKey::new(&key), chain);
-        let cryptde_enum = CryptDEEnum::Fake((main_cryptde, alias_cryptde));
-        Self::start_with_cryptde_enum(ports, index, host_node_parent_dir, cryptde_enum)
+impl MASQMockNodeWithMutableHandle {
+    pub fn absorb_configuration(&mut self, node_record: &NodeRecord) {
+        // Copy attributes from the NodeRecord into the MASQNode.
+        self.configurable_guts.earning_wallet = node_record.earning_wallet();
+        self.configurable_guts.rate_pack = *node_record.rate_pack();
     }
 
+    pub fn finalize_mutable_handle(self) -> MASQMockNode {
```

Comment:
Could this be an implementation of `Into<MASQMockNode>`?

If not, no worries; it just feels as though it'd be more Rustaceous.